src/codegen/compile.ml

```diff
@@ -4488,7 +4491,8 @@ module Serialization = struct
           (* Second time we see this *)
```

Comment:
I think the correct thing to do is to just remove the preceding Sanity checks as they will now have false positives.

Reply:
Maybe I'm confused, but I think those assertions are right. In the size calculation we replace mutable objects with `StableSeen`, so in this pass we shouldn't see mutable objects. All mutable objects should be turned into `StableSeen`s, or their headers should be updated with their offsets in the serialization buffer.